### DISCUSSION TITLE
Play background music when video is muted. Stop background music when video is unmuted

### DISCUSSION THREAD

**Discussion by frankusu** [TheWidlarzGroup/react-native-video#4156] [Q&A]
### What are you trying to do?

Keep background audio (Spotify) playing when the video is muted and playing, then stop the background audio when the video is unmuted and playing/paused.

### What have you tried?

**Expected**: Keep background audio (Spotify) playing when video is muted and playing -> stop background audio when video is unmuted and playing/paused

**Actual**: On iOS the music is mixed with audio no matter if muted or not. On Android the music always stops no matter if video is muted/unmuted. Right now it doesn't work for either iOS nor Android, and it's giving an error for Android saying the `onAudioFocusChanged` should return an object.

Looking at the react-native-video Swift code, there appears to be no `else if { mixWithOthers == "inherit" }` branch, so there seems to be no way to revert it back to inherit on iOS.

Already edited the AppDelegate as described in the ReadMe by adding the lines below.

`setIsVideoMuted` and `isVideoMuted` are passed in as props (code attached as screenshot in original post: https://github.com/user-attachments/assets/81e6a808-1db2-4de5-8458-78438189f559).

### Relevant code

react-native-video Swift code:

```tsx
        if ignoreSilentSwitch == "ignore" {
            category = audioOutput == "earpiece" ? AVAudioSession.Category.playAndRecord : AVAudioSession.Category.playback
        } else if ignoreSilentSwitch == "obey" {
            category = AVAudioSession.Category.ambient
        }

        if mixWithOthers == "mix" {
            options = .mixWithOthers
        } else if mixWithOthers == "duck" {
            options = .duckOthers
        }
```

AppDelegate edit:

```tsx
(BOOL)application:(UIApplication )application didFinishLaunchingWithOptions:(NSDictionary)launchOptions
{
  // Allow background music (ie. Spotify) to continue playing when app has video playing
  [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryAmbient error:nil];
```

### Version

v6

### Platforms

iOS, Android

**Comment by freeboub** [comment]
I have also seen some bad behavior this week, and the point you are pointing here!
I think the best would be to finish following PR which does some cleanup in ios audio management: https://github.com/TheWidlarzGroup/react-native-video/pull/3850

**Reply by frankusu** [reply]
thanks @freeboub will also post there

**Comment by freeboub** [comment]
@frankusu Did you try to add the `else if { mixWithOthers == "inherit" }` ?
I had a quick try, but I am not sure this patch is enough ...
If it fixes the issue, you can use it temporary with patch-package on your side and we will integrate it soon

**Reply by frankusu** [reply]
Hey @freeboub I think I managed to find a fix, let me clone and submit a PR

**Comment by frankusu** [comment]
`+++ b/node_modules/react-native-video/ios/Video/Features/RCTPlayerOperations.swift
@@ -155,12 +155,16 @@ enum RCTPlayerOperations {
             category = audioOutput == "earpiece" ? AVAudioSession.Category.playAndRecord : AVAudioSession.Category.playback
         } else if ignoreSilentSwitch == "obey" {
             category = AVAudioSession.Category.ambient
+        } else if ignoreSilentSwitch == "inherit" {
+            category = nil
         }
 
         if mixWithOthers == "mix" {
             options = .mixWithOthers
         } else if mixWithOthers == "duck" {
             options = .duckOthers
+        } else if mixWithOthers == "inherit" {
+            options = nil
         }
 
         if let category, let options {
@@ -198,6 +202,12 @@ enum RCTPlayerOperations {
             } catch {
                 debugPrint("[RCTPlayerOperations] Problem setting up AVAudioSession options. Error: \(error).")
             }
+        } else if category == nil, options == nil {
+            do {
+                try audioSession.setCategory(.playback, mode: .default)
+            } catch {
+                debugPrint("[RCTPlayerOperations] Problem setting AVAudioSession category to default. Error: \(error).")
+            }
         }
     }
 }`

**Reply by frankusu** [reply]
the formatting is way off, but with the highlighted lines with the * on the side, if you add those to a patch or PR it should work !

**Reply by freeboub** [reply]
Yes it is broken. Btw it looks good, can you please open a pr with this patch ?

**Comment by TWG-AI** [comment]
To keep background audio (e.g. Spotify) playing while a muted video plays on iOS, use `disableAudioSessionManagement={true}` on the `<Video />` component (v6) or set `player.disableAudioSessionManagement = true` in your setup callback (v7). This tells react-native-video to skip activating the `AVAudioSession`, so the system won't deactivate other apps' audio.

Note: setting `muted={true}` alone is not enough — the audio session is still activated on component mount. `disableAudioSessionManagement` is the correct prop for this use case.
